### ISSUE TITLE
Build Broken on Mac OSX

### ISSUE THREAD

**Issue by LeStarch** [nasa/fprime#299]
The (framework) build doesn't work on OSX.

```
-- Looking for pthread.h - not found
CMake Error at /Applications/CMake.app/Contents/share/cmake-3.5/Modules/FindPackageHandleStandardArgs.cmake:148 (message):
-- Configuring incomplete, errors occurred!
  Could NOT find Threads (missing: Threads_FOUND)
See also "/Users/mstarch/code/fprime-infra/fprime-sw/build-fprime-automatic-native/CMakeFiles/CMakeOutput.log".
Call Stack (most recent call first):
See also "/Users/mstarch/code/fprime-infra/fprime-sw/build-fprime-automatic-native/CMakeFiles/CMakeError.log".
  /Applications/CMake.app/Contents/share/cmake-3.5/Modules/FindPackageHandleStandardArgs.cmake:388 (_FPHSA_FAILURE_MESSAGE)
  /Applications/CMake.app/Contents/share/cmake-3.5/Modules/FindThreads.cmake:223 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  cmake/platform/Darwin.cmake:11 (FIND_PACKAGE)
  cmake/platform/CMakeLists.txt:31 (include)
  cmake/FPrime.cmake:22 (include)
  CMakeLists.txt:24 (include)
```

**Comment by LeStarch** [comment]
```
error: unknown warning option '-Wno-stringop-truncation'; did you mean '-Wno-string-conversion'?
```

**Comment by LeStarch** [comment]
Note: the first error is caused by the second.  We should also compare these flags with those in the  Darwin.cmake platform.

**Comment by Joshua-Anderson** [comment]
Ah, I forgot to test on Darwin after my fixes. The flags in Darwin.cmake are in addition to the ones in the root cmakelists file. I'll add the `-Wno-stringop-truncation` only for gcc. Do we care about the root cmake project being able to build in old versions of gcc (I.E. gcc 4)? This flag was added recently enough that it's not supported in super old GCC versions, but I can add a flag that tells the compiler to ignore unknown flags so it'll still build. But I think since this build is used only for CI and testing I'm not too worried about supporting old GCC

**Comment by LeStarch** [comment]
I am ok with it.  I would also test removing many of those Darwin flags...as why have them, they don't make a lot of sense.

**Comment by Joshua-Anderson** [comment]
Resolved by #301
